src/Arduino_LoRaConnectionHandler.cpp

```diff
@@ -0,0 +1,247 @@
+  modem.write(buf, size);
+  err = modem.endPacket(true);
+  /*Error manager according pr #68 of MKRWAN repo*/
+  if (err != size) {
+    switch (err) {
+      case -20: {
+          Serial.println("Message length is bigger than max LoRa packet!");
+        } break;
+      case -1: {
+          Serial.println("Message ack was not received, the message could not be delivered");
+        } break;
+      case -2: {
+          Serial.println("LoRa generic error (LORA_ERROR)");
+        } break;
+      case -3: {
+          Serial.println("LoRa malformed param error (LORA_ERROR_PARAM");
+        } break;
+      case -4: {
+          Serial.println("LoRa chip is busy (LORA_ERROR_BUSY)");
+        } break;
+      case -5: {
+          Serial.println("LoRa chip overflow error (LORA_ERROR_OVERFLOW)");
+        } break;
+      case -6: {
+          Serial.println("LoRa no network error (LORA_ERROR_NO_NETWORK)");
+        } break;
+      case -7: {
+          Serial.println("LoRa rx error (LORA_ERROR_RX)");
+        } break;
+      case -8: {
+          Serial.println("LoRa unknown error (LORA_ERROR_UNKNOWN)");
+        } break;
+    }
+  } else {
+    Serial.println("Message sent correctly!");
+  }
+  return err;
+}
+
+int LoRaConnectionHandler::read() {
+  return modem.read();
+}
+
+bool LoRaConnectionHandler::available() {
+  return modem.available();
+}
+
+void LoRaConnectionHandler::update() {
+
+  unsigned long const now = millis();
+  int networkStatus = 0;
+  if (now - lastConnectionTickTime > connectionTickTimeInterval) { /*  time bracket  */
+
+    lastConnectionTickTime = now;
+    switch (netConnectionState) {
+      case NetworkConnectionState::INIT:          netConnectionState = update_handleInit(); break;
+      case NetworkConnectionState::CONNECTING:    netConnectionState = update_handleConnecting(); break;
+      case NetworkConnectionState::CONNECTED:     netConnectionState = update_handleConnected(); break;
+      case NetworkConnectionState::DISCONNECTING: netConnectionState = update_handleDisconnecting(); break;
+      case NetworkConnectionState::DISCONNECTED:  netConnectionState = update_handleDisconnected(); break;
+      case NetworkConnectionState::ERROR:                                                            break;
+      case NetworkConnectionState::CLOSED:                                                           break;
+    }
+  }
+}
+
+/******************************************************************************
+   PRIVATE MEMBER FUNCTIONS
+ ******************************************************************************/
+
+NetworkConnectionState LoRaConnectionHandler::update_handleInit() {
+  Debug.print(DBG_VERBOSE, "::INIT");
+  if (!modem.begin(band)) {
+    Debug.print(DBG_VERBOSE, "Failed to start module");
+    execNetworkEventCallback(_on_error_event_callback, 0);
+    Debug.print(DBG_ERROR, "Something went wrong; are you indoor? Move near a window, then reset and retry.");
+  };
+  delay(1000);
```

Comment:
Do we need this delay call here?

Reply:
Yes, we found empirically that a delay is required in order to have a successful connection. The reason must be investigated at LoRa chip level. In one of the last commit I reduced the delay to 100ms

Reply:
If thats the case please create an issue in the LoRa repository with a minimal code example exhibiting the problematic behaviour.

Reply:
I've notified @facchinm about the behavior and he confirmed to me that the delay is required in certain conditions and it's a consequence of ADR. We think that it's better to not put any delay in the MKRWAN.h library in order to not impact every application. I resolve the conversation here because it is confirmed the delay is required in this case